src/main/kotlin/com/openlattice/mechanic/upgrades/SyncOrgEntitySetsMissingInMeta.kt

```diff
@@ -0,0 +1,118 @@
+
+            val orgEntitySetsIds = entitySetsService.getEntitySetsForOrganization(org.id)
+            if (orgEntitySetsIds.isEmpty()) {
+                logger.info("getEntitySetsForOrganization returned empty set")
+                return@forEachIndexed
+            }
+
+            val idPT = edmService.getPropertyType(ID_FQN)
+            if (idPT == null) {
+                logger.info("ol.id PropertyType is null")
+                return@forEachIndexed
+            }
+
+            val dataSetsESID = org.organizationMetadataEntitySetIds.datasets
+            val dsEntitySet = entitySetsService.getEntitySet(dataSetsESID)
+            if (dsEntitySet == null) {
+                logger.info("\"datasets\" meta entity set is null")
+                return@forEachIndexed
+            }
+
+            val data = dataService.getEntitySetData(
+                mapOf(dataSetsESID to Optional.empty()),
+                linkedSetOf(idPT.type.toString()),
+                mapOf(dataSetsESID to mapOf(idPT.id to idPT)),
+                dsEntitySet.isLinking
+            )
+
+            if (data.entities == null) {
+                logger.info("getEntitySetData returned null entities")
+                return@forEachIndexed
+            }
+
+            val ids = data.entities
+                .mapNotNull { entity -> entity }
+                .mapNotNull { entity ->
+                    val id = entity[ID_FQN]?.first()
+                    try {
+                        UUID.fromString(id.toString())
+                    } catch (e: Exception) {
+                        logger.error("invalid data set id $id", e)
+                        return@mapNotNull null
+                    }
+                }
+                .toSet()
+
+            val targetIds = orgEntitySetsIds.subtract(ids)
+            logger.info("found ${targetIds.size} entity sets to sync - $targetIds")
+
+            if (targetIds.isNotEmpty()) {
+                logger.info("starting syncing org entity sets")
+                syncOrgEntitySets(org, targetIds)
+                logger.info("finished syncing org entity sets")
+            }
+
+            logger.info("finished processing org ${org.id}")
+            logger.info("progress ${((index + 1) / orgs.size) * 100}%")
+            logger.info("================================")
+            logger.info("================================")
+        }
+
+        return true
+    }
+
+    private fun syncOrgEntitySets(org: Organization, ids: Set<UUID>) {
+        try {
+            logger.info("getting entity sets")
+            val entitySets = entitySetsService.getEntitySetsAsMap(ids).values
+            val propertyTypesByEntitySet = mutableMapOf<UUID, Collection<PropertyType>>()
+            logger.info("getting property types")
+            entitySets.forEach { entitySet ->
+                val propertyTypes = edmService.getPropertyTypesOfEntityType(entitySet.entityTypeId)
```

Comment:
this is going to be suuuper slow since it's going to make two separate hazelacst calls per entity set, and doesn't actually need to interact with hazelcast at all. The EDM doesn't differ by entity set, so you should just use the EDM values that are cached in memory through toolbox.